db/changes.go

```diff
@@ -649,25 +666,26 @@ func (db *Database) SimpleMultiChangesFeed(ctx context.Context, chans base.Set,
 
 		// Restrict to available channels, expand wild-card, and find since when these channels
 		// have been available to the user:
-		var channelsSince channels.TimedSet
+		channelsSince := channels.CollectionByTimedSet{}
 		if db.user != nil {
-			var channelsRemoved []string
-			channelsSince, channelsRemoved = db.user.FilterToAvailableChannels(chans)
+			chansSince, channelsRemoved := db.user.FilterToAvailableChannels(chans)
 			if len(channelsRemoved) > 0 {
 				base.InfofCtx(ctx, base.KeyChanges, "Channels %s request without access by user %s", base.UD(channelsRemoved), base.UD(db.user.Name()))
 			}
+			channelsSince[singleCollectionID] = chansSince
 		} else {
-			channelsSince = channels.AtSequence(chans, 0)
+			channelsSince = channels.AtSequenceByCollection(chans, 0)
 		}
 
 		// Mark channel set as active, schedule defer
-		db.activeChannels.IncrChannels(channelsSince)
-		defer db.activeChannels.DecrChannels(channelsSince)
+		activeChannels := channelsSince.GetChannels()
```

Comment:
I did this, but I feel like we are still iterating, we just move where we do it. I am not sure I can see the 100% picture without the principal changes. I left this as a separate commit in case you are indecisive also.

Reply:
Aren't we just iterating over collections and channels once now?  Previously we were iterating over collections and channels to build a slice of ID, then iterating again over that slice of IDs to Incr/Decr channels.

Reply:
We do iterate over these to build the TimedSetByCollection, and there is one fewer iteration than before.